src/mongo_client.ts

```diff
@@ -446,6 +452,14 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
     forceOrCallback?: boolean | Callback<void>,
     callback?: Callback<void>
   ): Promise<void> | void {
+    // There's no way to set hasBeenClosed back to false
+    Object.defineProperty(this.s, 'hasBeenClosed', {
```

Comment:
There was discussion in driver devs about whether or not we should allow clients to automatically reconnect if the client is closed ([thread](https://mongodb.slack.com/archives/C72LB5RPV/p1652213273714369)).  Based on the feedback, we decided no.